src/runtime_src/core/tools/xbutil2/SubCmdReset.cpp

```diff
@@ -92,9 +120,23 @@ SubCmdReset::execute(const SubCmdOptions& _options) const
   }
 
   // -- Now process the subcommand --------------------------------------------
+  // Collect all of the devices of interest
+  std::set<std::string> deviceNames;
+  xrt_core::device_collection deviceCollection;  // The collection of devices to examine
+  for (const auto & deviceName : devices) 
+    deviceNames.insert(boost::algorithm::to_lower_copy(deviceName));
+
+  XBU::collect_devices(deviceNames, true /*inUserDomain*/, deviceCollection);
+  pretty_print_action_list(deviceCollection, reset);
 
-  XBU::error("COMMAND BODY NOT IMPLEMENTED.");
-  // TODO: Put working code here
+  // Ask user for permission
+  if(!XBU::can_proceed())
+    return;
+
+  //perform reset actions
+  for (const auto & dev : deviceCollection) {
+    reset_device(dev, reset);
```

Comment:
The code should not be passing reset string names.  It should instead be using enumeration types. 

Please create:

1. An enumeration type to handle all of the reset values.
2. Create two methods to translate "strings to the enumeration type" and "enumeration type to strings".  
3. Update xbmgmt's code base to use the enumeration types as well.  

Note: Most likely this enumeration type and helper methods will be located in core/tools/common.

Reply:
That's a good idea. I created enum class in XBUtilities and created a string to enum function which uses a map to do the conversion. I don't need the enum to string method.